ali atapi dma is broken right now. disable it (but not disk dma)
until the problem is better understood.

--- 1/drivers/ata/pata_ali.c~	2007-05-04 18:22:41.129254272 +0100
+++ 2/drivers/ata/pata_ali.c	2007-05-04 18:22:57.105825464 +0100
@@ -329,6 +329,11 @@
 	adev->max_sectors = 255;
 }
 
+static int ali_no_atapi_dma(struct ata_queued_cmd *qc)
+{
+	return -1;
+}
+
 static struct scsi_host_template ali_sht = {
 	.module			= THIS_MODULE,
 	.name			= DRV_NAME,
@@ -371,6 +376,7 @@
 
 	.qc_prep 	= ata_qc_prep,
 	.qc_issue	= ata_qc_issue_prot,
+	.check_atapi_dma= ali_no_atapi_dma,
 
 	.data_xfer	= ata_data_xfer,
 
@@ -412,6 +418,7 @@
 
 	.qc_prep 	= ata_qc_prep,
 	.qc_issue	= ata_qc_issue_prot,
+	.check_atapi_dma= ali_no_atapi_dma,
 
 	.data_xfer	= ata_data_xfer,
 
@@ -450,6 +457,7 @@
 
 	.qc_prep 	= ata_qc_prep,
 	.qc_issue	= ata_qc_issue_prot,
+	.check_atapi_dma= ali_no_atapi_dma,
 
 	.data_xfer	= ata_data_xfer,
 
@@ -487,6 +495,7 @@
 
 	.qc_prep 	= ata_qc_prep,
 	.qc_issue	= ata_qc_issue_prot,
+	.check_atapi_dma= ali_no_atapi_dma,
 
 	.data_xfer	= ata_data_xfer,
 

